docs/DEVELOP.md

```diff
@@ -1,13 +1,75 @@
 # Development Guide for rdiff-backup
 
-Some notes for developers and other people willing to help development,
-or simply compile rdiff-backup from source code.
+> Suggest [improvements to this documentation](https://github.com/rdiff-backup/rdiff-backup/issues/new?title=Docs%20feedback:%20/docs/DEVELOP.md)!
 
 
-## PRE-REQUISITES
+## GETTING THE SOURCE
 
-The same pre-requisites as for the installation of rdiff-backup are
-required as well:
+Simply clone the source with:
+
+	git clone https://github.com/rdiff-backup/rdiff-backup.git
+
+> **NOTE:** If you plan to provide your own code, you should first fork
+our repo and clone your own forked repo. How is described at
+https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/working-with-forks
+
+
+## GENERAL GUIDELINES
+
+- Before committing to a lot of writing or coding, please file an issue on Github and discuss your plans and gather feedback. Eventually it will be much easier to merge your change request if the idea and design has been agreed upon, and there will be less work for you as a contributor if you implement your idea along the correct lines to begin with.
+- Please check out [existing issues](https://github.com/rdiff-backup/rdiff-backup/issues) and [existing merge requests](https://github.com/rdiff-backup/rdiff-backup/pulls) and browse the [git history](https://github.com/rdiff-backup/rdiff-backup/commits/master) to see if somebody already tried to address the thing you have are interested in. It might provide useful insight why the current state is as it is.
+- Changes can be submitted using the typical Github workflow: clone this repository, make your changes, test and verify, and submit a Pull Request.
+- Each change (= pull request) should focus on some topic and resist changing anything else. Keeping the scope clear also makes it easier to review the pull request. A good pull request has only one or a few commits, with each commit having a good commit subject and if needed also a body that explains the change.
+- For all code changes, please remember also to include inline comments and update tests where needed.
+
+### License
+
+Rdiff-backup is released as GNU General Public License v2.0. By contributing to
+this repository you agree that your work is licensed using the chosen project
+license.
+
+### Branching model and pull requests
+
+The *master* branch is always kept in a clean state. Anybody can at any time
```

Comment:
Let me re-comment at the right place: anybody can fork and provide a pull request, only selected developers can fork.